tests/ui/duplicate_entry_error.stderr

```diff
@@ -1,8 +1,11 @@
 error[E0152]: found duplicate lang item `panic_impl`
   --> $DIR/duplicate_entry_error.rs:11:1
    |
-LL | fn panic_impl(info: &PanicInfo) -> ! {
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+LL | / fn panic_impl(info: &PanicInfo) -> ! {
```

Comment:
The old `def_span` logic is supposedly reproduced in this PR, so why do some spans change?

Reply:
I don't reproduce it fully -- I'm just caching the item span here.

In HIR, we do some special span creation to essentially compute the "head span" of each item:
https://github.com/rust-lang/rust/blob/ff2c56344c764af598ad33027e9c7a48881808ef/compiler/rustc_middle/src/hir/map/mod.rs#L839-L948